magmi/plugins/extra/itemprocessors/customoptions/pablo_customoptions.php

```diff
@@ -121,7 +121,7 @@ public function createOptionValues($field, $sids, $valarr)
         $cvalarr = count($valarr);
         for ($i = 0; $i < $cvalarr; $i++) {
             $val = $valarr[$i];
-            if ($i < count($optionTypeIds)) {
+            if ($optionTypeIds && $i < count($optionTypeIds)) {
```

Comment:
what about changing the default value to empty array in https://github.com/macopedia/magmi-m2/blob/58a7d9769edf5759eb00f1f358c5dbed2085ef9c/magmi/plugins/extra/itemprocessors/customoptions/pablo_customoptions.php#L39 ?
like:
```
        return isset($this->_opttypeids[$field]) ? $this->_opttypeids[$field] : [];

```

Reply:
Actually i did that first too, but i thought maybe an empty array could be unexpected somewhere else. I didn't look much into it, i'm okay with both solutions :) What do you prefer?

Reply:
I've checked and this method is used in the single place, so its safe to change it

Reply:
Okay, i changed the PR